### ISSUE TITLE
Equivalent upload file function similar in rest-client upload

### ISSUE THREAD

**Issue by Eric-Guo** [httprb/http#293]
I try to migrate from Rest Client to http, but don't known how to write proper [upload options similar to rest-client](https://github.com/rest-client/rest-client#usage-raw-url):

In rest client, I can write:

```ruby
RestClient.post( url, { upload: { media: File.new(path, 'rb') } })
```

But in http, tried several combination, even relative raw one:

```ruby
form = HTTP::FormData.create(media: HTTP::FormData::File.new(file_path)).to_s
HTTP.post url, body: "#{form}\r\n"
```

But seems still not exactly same one.

**Comment by tarcieri** [comment]
What's the specific problem? The request being made is different?

**Comment by ixti** [comment]
Below should work:

``` ruby
HTTP.post(url, :form => {
   :upload => { :media => HTTP::FormData::File.new(path) }
  }
})
```

**Comment by tarcieri** [comment]
Please reopen if you're still having problems

**Comment by Eric-Guo** [comment]
Following ixti suggest, written as [below](https://github.com/Eric-Guo/wechat/commit/d426dc9490f47d3dde7ca6894bc67fab5d4dc70c), meet another ` failed to read headers` error

```ruby
    def post_file(path, file, post_header = {})
      request(path, post_header) do |url, header|
        params = header.delete(:params)
        HTTP.headers(header).post(url, params: params, form: { media: HTTP::FormData::File.new(file) }, ssl_context: ssl_context)
      end
    end
```



```bat
D:\>wechat material_add image logo.jpg
C:/Ruby22/lib/ruby/gems/2.2.0/gems/http-1.0.1/lib/http/connection.rb:110:in `rescue in read_headers!': failed to read headers: EOFError (HTTP::ConnectionError)
        from C:/Ruby22/lib/ruby/gems/2.2.0/gems/http-1.0.1/lib/http/connection.rb:99:in `read_headers!'
        from C:/Ruby22/lib/ruby/gems/2.2.0/gems/http-1.0.1/lib/http/client.rb:66:in `perform'
        from C:/Ruby22/lib/ruby/gems/2.2.0/gems/http-1.0.1/lib/http/client.rb:43:in `request'
        from C:/Ruby22/lib/ruby/gems/2.2.0/gems/http-1.0.1/lib/http/chainable.rb:25:in `post'
        from C:/Ruby22/lib/ruby/gems/2.2.0/gems/wechat-0.7.0/lib/wechat/client.rb:31:in `block in post_file'
        from C:/Ruby22/lib/ruby/gems/2.2.0/gems/wechat-0.7.0/lib/wechat/client.rb:41:in `request'
        from C:/Ruby22/lib/ruby/gems/2.2.0/gems/wechat-0.7.0/lib/wechat/client.rb:29:in `post_file'
        from C:/Ruby22/lib/ruby/gems/2.2.0/gems/wechat-0.7.0/lib/wechat/api_base.rb:39:in `block in post_file'
        from C:/Ruby22/lib/ruby/gems/2.2.0/gems/wechat-0.7.0/lib/wechat/api_base.rb:45:in `with_access_token'
        from C:/Ruby22/lib/ruby/gems/2.2.0/gems/wechat-0.7.0/lib/wechat/api_base.rb:38:in `post_file'
        from C:/Ruby22/lib/ruby/gems/2.2.0/gems/wechat-0.7.0/lib/wechat/corp_api.rb:155:in `material_add'
        from C:/Ruby22/lib/ruby/gems/2.2.0/gems/wechat-0.7.0/bin/wechat:337:in `material_add'
        from C:/Ruby22/lib/ruby/gems/2.2.0/gems/thor-0.19.1/lib/thor/command.rb:27:in `run'
        from C:/Ruby22/lib/ruby/gems/2.2.0/gems/thor-0.19.1/lib/thor/invocation.rb:126:in `invoke_command'
        from C:/Ruby22/lib/ruby/gems/2.2.0/gems/thor-0.19.1/lib/thor.rb:359:in `dispatch'
        from C:/Ruby22/lib/ruby/gems/2.2.0/gems/thor-0.19.1/lib/thor/base.rb:440:in `start'
        from C:/Ruby22/lib/ruby/gems/2.2.0/gems/wechat-0.7.0/bin/wechat:426:in `<top (required)>'
        from C:/Ruby22/bin/wechat:23:in `load'
        from C:/Ruby22/bin/wechat:23:in `<main>'
```

**Comment by ixti** [comment]
I'm not 100% sure, but isn't RestClient adds `Accept: application/json` header?
Try add it yourself: `headers[:accept] = "application/json"`

**Comment by ixti** [comment]
Ahh.. Opened `client.rb` of wechat gem. You are merging in `:accept => :json` which is not automagically translated into `application/json` in HTTP as opposed to RestClient which tries to resolve given value via MimeType treating value as extension.

# TLDR

Change [client.rbL40](https://github.com/Eric-Guo/wechat/blob/d426dc9490f47d3dde7ca6894bc67fab5d4dc70c/lib/wechat/client.rb#L40) to:


``` ruby
header.merge!(accept: "applicaiton/json")
```

**Comment by Eric-Guo** [comment]
`header.merge!('Accept' => 'application/json')` still have below problem, I can confirm the uploading is good, except the `read_headers!` alwasy failed.

```
C:/Ruby22/lib/ruby/gems/2.2.0/gems/http-1.0.1/lib/http/connection.rb:110:in `rescue in read_headers!': failed to read headers: EOFError (HTTP::ConnectionError)
        from C:/Ruby22/lib/ruby/gems/2.2.0/gems/http-1.0.1/lib/http/connection.rb:99:in `read_headers!'
        from C:/Ruby22/lib/ruby/gems/2.2.0/gems/http-1.0.1/lib/http/client.rb:66:in `perform'
        from C:/Ruby22/lib/ruby/gems/2.2.0/gems/http-1.0.1/lib/http/client.rb:43:in `request'
        from C:/Ruby22/lib/ruby/gems/2.2.0/gems/http-1.0.1/lib/http/chainable.rb:25:in `post'
        from C:/Ruby22/lib/ruby/gems/2.2.0/gems/wechat-0.7.0/lib/wechat/client.rb:31:in `block in post_file'
        from C:/Ruby22/lib/ruby/gems/2.2.0/gems/wechat-0.7.0/lib/wechat/client.rb:41:in `request'
        from C:/Ruby22/lib/ruby/gems/2.2.0/gems/wechat-0.7.0/lib/wechat/client.rb:29:in `post_file'
        from C:/Ruby22/lib/ruby/gems/2.2.0/gems/wechat-0.7.0/lib/wechat/api_base.rb:39:in `block in post_file'
        from C:/Ruby22/lib/ruby/gems/2.2.0/gems/wechat-0.7.0/lib/wechat/api_base.rb:45:in `with_access_token'
        from C:/Ruby22/lib/ruby/gems/2.2.0/gems/wechat-0.7.0/lib/wechat/api_base.rb:38:in `post_file'
        from C:/Ruby22/lib/ruby/gems/2.2.0/gems/wechat-0.7.0/lib/wechat/corp_api.rb:155:in `material_add'
        from C:/Ruby22/lib/ruby/gems/2.2.0/gems/wechat-0.7.0/bin/wechat:337:in `material_add'
        from C:/Ruby22/lib/ruby/gems/2.2.0/gems/thor-0.19.1/lib/thor/command.rb:27:in `run'
        from C:/Ruby22/lib/ruby/gems/2.2.0/gems/thor-0.19.1/lib/thor/invocation.rb:126:in `invoke_command'
        from C:/Ruby22/lib/ruby/gems/2.2.0/gems/thor-0.19.1/lib/thor.rb:359:in `dispatch'
        from C:/Ruby22/lib/ruby/gems/2.2.0/gems/thor-0.19.1/lib/thor/base.rb:440:in `start'
        from C:/Ruby22/lib/ruby/gems/2.2.0/gems/wechat-0.7.0/bin/wechat:426:in `<top (required)>'
        from C:/Ruby22/bin/wechat:23:in `load'
        from C:/Ruby22/bin/wechat:23:in `<main>'
```

**Comment by tarcieri** [comment]
An EOFError indicates the remote server is closing the connection
